ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -14495,6 +14495,9 @@ private void validateStorageFormat(
                           + "'='" + fileFormat + "')");
         }
       }
+    } else {
+      throw new SemanticException("Partition transform is only supported for specific storage handler tables. e.g. " +
```

Comment:
If you think the exception hint is not good, please feel free to give your opinion here.
Thanks.

Reply:
make sense, I think we could be more specific "Partition transforms are only supported by Iceberg storage handler"

Reply:
Changed.

Reply:
@zhangbutao, should we add this to ErrorMsg?

Reply:
Done. Added exception to ErrorMsg.